packages/client/components/TaskFooterIntegrateMenuList.tsx

```diff
@@ -110,10 +110,12 @@ const TaskFooterIntegrateMenuList = (props: Props) => {
 
   useEffect(() => {
     // if searching for a repoIntegration that doesnt exist in the cache, it could be stale so use the network
+    // It is possible that user has a lot of repositories, if nothing was found in initial request, query everything
+    const veryBigNumberOfRepositories = 10000
     if (!networkOnly && filteredIntegrations.length === 0) {
       setNetworkOnly(true)
       setKeepParentFocus(false)
-      setFirst((first) => first + 50)
+      setFirst((first) => first + veryBigNumberOfRepositories)
```

Comment:
+1 the correct solution would be to check the `hasMore` field and fetch everything by increasing `after` until we fetched all candidates. Only for the first fetch we should set `networkOnly = true` as then the cached result on the server would be up to date already.
I think this solution is good enough for now, so let's not invest the additional work now.